js/damage.js

```diff
@@ -629,7 +629,7 @@ function getDamageResult(attacker, defender, move, field) {
 		finalMods.push(field.format !== "Singles" ? (gen >= 6 ? 0xAAC : 0xA8F) : 0x800);
 		description.isLightScreen = true;
 	}
-	if ((defAbility === "Multiscale" || defAbility === "Shadow Shield") && defender.curHP === defender.maxHP && !field.isSR && !field.spikes) {
+	if ((defAbility === "Multiscale" || defAbility === "Shadow Shield") && defender.curHP === defender.maxHP && !field.isSR && (!field.spikes || (defAbility === "Multiscale" && field.spikes && [defender.type1, defender.type2].indexOf("Flying") != -1))) {
```

Comment:
You don't need `defAbility === "Multiscale" &&`

Reply:
Ah yes, my bad. Fixing it right away.

Reply:
The entirety of

`(!field.spikes || (defAbility === "Multiscale" && field.spikes && [defender.type1, defender.type2].indexOf("Flying") != -1))`

is overcomplicated, really.

`(!field.spikes || [defender.type1, defender.type2].indexOf("Flying") != -1)`

does the exact same thing, although you can optimize a bit with

`(!field.spikes || defender.type1 === "Flying" || defender.type2 === "Flying")`